pkg/apis/provisioning/v1alpha2/provisioner.go

```diff
@@ -58,12 +57,14 @@ type ProvisionerSpec struct {
 // Cluster configures the cluster that the provisioner operates against. If
 // not specified, it will default to using the controller's kube-config.
 type Cluster struct {
-	// Name is required to detect implementing cloud provider resources.
-	// +required
-	Name string `json:"name"`
-	// CABundle is required for nodes to verify API Server certificates.
-	// +required
-	CABundle string `json:"caBundle"`
+	// Name may be required to detect implementing cloud provider resources.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// CABundle used by nodes to verify API Server certificates. If omitted,
+	// it will be dynamically loaded at runtime from the in-cluster configuration
+	// file /var/run/secrets/kubernetes.io/serviceaccount/ca.crt.
+	// +optional
+	CABundle string `json:"caBundle,omitempty"`
 	// Endpoint is required for nodes to connect to the API Server.
 	// +required
 	Endpoint string `json:"endpoint"`
```

Comment:
Consider moving endpoint to the top of these three fields, now that it's the only required one.